src/DatatypeChannels.ASB/Consumer.fs

```diff
@@ -114,8 +122,7 @@ let mkNew (options: Options)
                 | Some (receiver, _, msgCtxs) ->
                     for KeyValue(_,msgCtx) in msgCtxs.ToArray() do
                         msgCtx.Close()
-                    msgCtxs.Clear()
-                    receiver.DisposeAsync().AsTask().Wait()
+                    receiver.CloseAsync().Wait()
```

Comment:
According to the docs the dispose is identical to close, but close has slightly nicer signature.